spec/data-model/README.md

```diff
@@ -110,38 +110,45 @@ interface CatchallKey {
 ## Patterns
 
 Each `Pattern` represents a linear sequence, without selectors.
-Each element of the `body` array MUST either be a non-empty string or an `Expression` object.
+Each element of the `body` array MUST either be a non-empty string or an `Expression` or `Markup` object.
 String values represent literal _text_,
-while `Expression` wraps each of the potential _expression_ shapes.
+while `Expression` and `Markup` wrap each of the potential _expression_ and _markup_ shapes.
 The `body` strings are the "cooked" _text_ values, i.e. escape sequences are processed.
 
-Implementations MUST NOT rely on the set of `Expression` interfaces being exhaustive,
+Implementations MUST NOT rely on the set of `Expression` and `Markup` interfaces being exhaustive,
 as future versions of this specification MAY define additional expressions.
```

Comment:
I think the term "expression" is overloaded here. Also, the second normative word shouldn't be normative (it's permitting a future version of this specification to do something). Perhaps:

```suggestion
Implementations MUST NOT rely on the set of `Expression` and 
`Markup` interfaces defined in this document being exhaustive.
Future versions of this specification might define additional
types of expression or markup.
```